translate.js

```diff
@@ -0,0 +1,70 @@
+function translateJsonCompiler (output) {
+  var ret = {};
+
+  ret['errors'] = [];
+  for (var error in output['errors']) {
+    // FIXME: parse warnings here
+    ret['errors'].push({
+      type: 'Error',
+      component: 'general',
+      severity: 'error',
+      message: output['errors'][error],
+      formattedMessage: output['errors'][error]
+    });
+  }
+
+  ret['contracts'] = {}
+  for (var contract in output['contracts']) {
+    // Split name first, can be `contract`, `:contract` or `filename:contract`
+    var tmp = contract.match(/^([^:]*):?([^:]+)$/)
+    var fileName, contractName
+    if (tmp.length == 3) {
```

Comment:
So the match results will be:
- first match is filename without `:`
- second match is filename with `:`
- third match if contractname?